### PR TITLE
add (docker-)compose example

### PR DESCRIPTION
**Pull request by violoncelloCH**
This adds a compose.yml sample config to easily deploy a slidingsync proxy accompanied by a postgresql db container.

### Pull Request Checklist

- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)

### CODE REVIEW

File: compose.yml
Comment (kegsay) [inline compose.yml:28]: Why?

File: compose.yml
Comment (kegsay) [inline compose.yml:8]: Probably should have a versioned tag.

**Review by kegsay — CHANGES_REQUESTED** [review]

File: compose.yml
Comment (violoncelloCH) [reply compose.yml:24]: ohh sure, I'll add that!

File: compose.yml
Comment (violoncelloCH) [reply compose.yml:28]: whoops, didn't notice - leftover from the template I copied... I'll remove it

File: compose.yml
Comment (violoncelloCH) [reply compose.yml:8]: this means that it has to be incremented regularly, but it makes sense obviously to have it... though I was unsure to how to handle this before a stable version of the project. As soon as there are major versions I guess we can simply pin it to a major version...
should I add an exact version tag for now ?

File: compose.yml
Comment (violoncelloCH) [reply compose.yml:12]: I'm actually using this in production, but I guess it can suit well for development too...
I'd leave it up to you to if you as project maintainer if you want to include a pre-generated secret for a dev-setup that would have to be changed for production use or leave the field empty.
for defaulting to matrix.org: I did this as the Readme defaults to matrix.org as well (docker command in Readme section - that's what I've used as a base to build this compose.yml)... I personally use this with my own homeserver to test-drive Element X... let me know if you want me to remove the default :)

File: compose.yml
Comment (violoncelloCH) [reply compose.yml:16]: thanks for the recommendation! I'll change this :)

**Comment by violoncelloCH** [comment]
thanks for the review @kegsay ! and sorry for the slow response time - have been busy with university work the last weeks...

**Review by kegsay — COMMENTED** [review]
I remain concerned that this is in a weird place where it doesn't quite target production deployments but also doesn't quite target testing deployments.

If we want to have this targeting production deployments then we need more parameterisation and some CI which runs this to ensure it doesn't break. 

If we want this to target testing deployments then some comments around things to change (version tag, secret, postgres password) should be all that's left.

I'm leaning towards targeting testing deployments so people can just `docker-compose up` and get a working sliding sync instance. This means we need:
 - to mention that you can `docker-compose up` in the README
 - add comments around the things which aren't safe to have in production.

**Comment by kegsay** [comment]
Closing this due to inactivity. Please open another PR if you wish to target testing deployments.

I'm not opposed to having a docker-compose for simplicity, but we must be clear who the target audience is so we can make sure the defaults/exposed configuration are appropriate.

**Pull request closed without merging**
